jbi/services/jira.py

```diff
@@ -103,17 +102,55 @@ def raise_for_status(self, *args, **kwargs):
     get_project = instrumented_method(Jira.get_project)
 
     @instrumented_method
-    def permitted_projects(self, permissions: Iterable):
+    def paginated_projects(
+        self,
+        included_archived=None,
+        expand=None,
+        url=None,
+        keys: Optional[Collection[str]] = None,
+    ):
+        """Returns a paginated list of projects visible to the user.
+
+        https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-projects/#api-rest-api-2-project-search-get
+
+        We've patched this method of the Jira client to accept the `keys` param.
+        """
+
+        if not self.cloud:
+            raise ValueError(
+                "``projects_from_cloud`` method is only available for Jira Cloud platform"
+            )
+
+        params = {}
+
+        if keys is not None:
+            if len(keys) > 50:
```

Comment:
Do we care to have this check in two places now? Is this trying to safeguard us in case we change where they array items come from in the future?

Reply:
> Is this trying to safeguard us in case we change where they array items come from in the future?

Yep, if/when we eventually use this function to return paginated results for more than 50 projects, we'll still want to ensure `keys < 50`. Say we had 100 configured projects -- this guard would be a reminder for us to use two requests with 50 keys each, since the API only allows 50 keys per request ([docs](https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-projects/#api-rest-api-2-project-search-get)).